basic.go

```diff
@@ -93,67 +93,83 @@ func (b *basicBus) tryDropNode(typ reflect.Type) {
 	b.lk.Unlock()
 }
 
+type sub struct {
+	ch      chan interface{}
+	nodes   []*node
+	dropper func(reflect.Type)
+}
+
+func (s *sub) Out() <-chan interface{} {
+	return s.ch
+}
+
+func (s *sub) Close() error {
+	close(s.ch)
```

Comment:
we should probably make `s.ch` nil to allow for gc to vacuum the channel even if the consumer is retaining a reference to the sub.